### PR TITLE
PR to fix failed automatic update of eslint to v9.6.0 #4688

### PR DESCRIPTION
**Pull request by anasKhafaga**

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -19,6 +19,8 @@
     "npm": ">=7"
   },
   "dependencies": {
+    "@eslint/eslintrc": "^3.1.0",
+    "@eslint/js": "^9.6.0",
```
Comment (shiftkey) [inline package.json]: 🤔 Should these also be `devDependencies`?

File: eslint.config.mjs [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+        "func-names": "off",
+        "one-var": "off",
+        eqeqeq: "off",
+        "no-param-reassign": "off",
+        "no-underscore-dangle": "off",
+        "no-useless-escape": "off",
+        "no-restricted-globals": "off",
+        "no-use-before-define": "off",
+        "global-require": "off",
+        "import/no-amd": "off",
```
Comment (shiftkey) [inline eslint.config.mjs]: 🤔 This rule is from `eslint-plugin-import` so I'd expect some sort of error if we've had to remove the plugin...

File: package.json (hunk 1 above)
Comment (anasKhafaga) [reply package.json]: That's my bad, I pushed more focus to migration of `eslint` configuration. It's fixed now.

File: eslint.config.mjs (hunk 2 above)
Comment (anasKhafaga) [reply eslint.config.mjs]: Actually there was no thrown error so I haven't noticed it. But you are right, there's no need for it. Removed!

Thanks for your review @shiftkey

**Review by shiftkey — APPROVED** [review]

**Pull request merged**